client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -117,6 +124,21 @@ public void readerOffline(String readerId, Position lastPosition) {
         return synchronizer.getState().getStreamNames();
     }
 
+    public ReaderGroupState updateConfigState(ReaderGroupState.ConfigState configState, ReaderGroupState state, AtomicBoolean b) {
```

Comment:
rename `b`. add a javadoc on this method explaining what this is doing. i think this is not exposed on the interface ReaderGroup but this is a public method on the impl object. 

In fact, instead of passing AtomicBoolean in this method's signature, we should return boolean as the return param from this method. 

something like this:
```
boolean updateConfigState(ReaderGroupState.ConfigState configState, ReaderGroupState state) { 
       AtomicBoolean successfullyUpdated = new AtomicBoolean();
        synchronizer.updateState((s, updates) -> {
            successfullyUpdated .set(s.equals(state));
            if (successfullyUpdated.get()) {
                updates.add(new ChangeConfigState(configState, s.getGeneration() + 1));
            }
        });
        return successfullyUpdated.get();
    }
```

The caller can always get the state from RG object. right?

Reply:
I've renamed `b`. This method is now `private`. State updates for any operation on the RG (create, delete, reset) happen within `ReaderGroupImpl`. I've added comments explaining the reasoning behind the boolean.